hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterRpcClient.java

```diff
@@ -1175,6 +1177,7 @@ public <R extends RemoteLocationContext, T> RemoteResult invokeSequential(
     }
     // Return the first result, whether it is the value or not
     @SuppressWarnings("unchecked") T ret = (T) firstResult;
+    @SuppressWarnings("unchecked") R loc = (R) firstLocation;
     return new RemoteResult<>(locations.get(0), ret);
```

Comment:
Hi, @hfutatzhanghb According to your explanation, maybe this should be changed to:
```java
    @SuppressWarnings("unchecked") R loc = (R) firstLocation;
    return new RemoteResult<>(loc, ret);
```